nomad/acl_endpoint.go

```diff
@@ -1716,7 +1716,15 @@ func (a *ACL) UpsertAuthMethods(
 	}
 
 	// Validate each auth method, canonicalize, and compute hash
+	// merge methods in case we're doing an update
 	for idx, authMethod := range args.AuthMethods {
+		// if there's an existing method with the same name, we treat this as
+		// an update
+		existingMethod, _ := stateSnapshot.GetACLAuthMethodByName(nil, authMethod.Name)
```

Comment:
`GetACLAuthMethodByName` will only error when it can't find the existing method, what matters to us is that it will always return nil if it can't find the method.